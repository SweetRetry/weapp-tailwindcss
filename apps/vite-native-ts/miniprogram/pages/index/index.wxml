<view class="p-4">
  <IceButton class="bg-[#d72929]" my-class="bg-[orange]"></IceButton>
</view>