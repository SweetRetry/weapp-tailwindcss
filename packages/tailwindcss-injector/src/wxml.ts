import type { Buffer } from 'node:buffer'
import path from 'node:path'
import fs from 'fs-extra'
import { Parser } from 'htmlparser2'
import MagicString from 'magic-string'

const srcImportTagsMap: Record<string, string[]> = {
  // audio: ['src', 'poster'],
  // video: ['src', 'poster'],
  // image: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxs/01wxs-module.html
  // wxs: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxml/import.html
  import: ['src'],
  include: ['src'],
}

export interface WxmlDep {
  tagName: string
  start: number
  end: number
  quote: string | null | undefined
  name: string
  value: string
  attrs: Record<string, string>
}

// https://github.com/fb55/htmlparser2/issues/1541
// https://developers.weixin.qq.com/miniprogram/dev/framework/view/wxml/event.html
export function processWxml(wxml: string | Buffer) {
  const ms = new MagicString(wxml.toString())
  const deps: WxmlDep[] = []
  let currentTagName = ''
  let importAttrs: undefined | string[]
  let attrs: Record<string, string> = {}

  const parser = new Parser(
    {
      onopentagname(name) {
        currentTagName = name
        importAttrs = srcImportTagsMap[currentTagName]
      },
      onattribute(name, value, quote) {
        attrs[name] = value
        if (importAttrs) {
          for (const attrName of importAttrs) {
            if (attrName === name) {
              deps.push({
                name,
                value,
                quote,
                tagName: currentTagName,
                start: parser.startIndex,
                end: parser.endIndex,
                attrs,
              })
            }
          }
        }
      },
      onclosetag() {
        currentTagName = ''
        attrs = {}
        importAttrs = undefined
      },
    },
    {
      lowerCaseTags: false,
      xmlMode: true,
    },
  )
  parser.write(
    ms.original,
  )
  parser.end()
  return deps
}

export async function getDepFiles(file: string) {
  const set = new Set<string>()

  async function getDep(filepath: string) {
    if (await fs.exists(filepath)) {
      const dirname = path.dirname(filepath)
      set.add(filepath)
      const wxml = await fs.readFile(filepath, 'utf8')
      const deps = await processWxml(wxml)
      for (const { value } of deps) {
        const p = path.resolve(dirname, value)
        if (!set.has(p)) {
          await getDep(p)
        }
      }
    }
  }

  await getDep(file)

  return set
}
