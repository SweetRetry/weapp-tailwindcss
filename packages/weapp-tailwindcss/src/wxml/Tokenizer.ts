// 针对微信的属性表达式进行切词
// https://github.com/sonofmagic/weapp-tailwindcss/issues/319
// all state
export enum State {
  START,
  TEXT,
  OPEN_BRACE,
  POTENTIAL_CLOSE,
  BRACES_COMPLETE,
}

export class TokenizerStateMachine {
  private state: State
  private buffer: string
  private tokens: string[]

  constructor() {
    this.state = State.START
    this.buffer = ''
    this.tokens = []
  }

  private processChar(char: string) {
    switch (this.state) {
      case State.START:
        if (char === ' ') {
          // Ignore leading spaces
        }
        else if (char === '{') {
          this.state = State.OPEN_BRACE
          this.buffer += char
        }
        else {
          this.state = State.TEXT
          this.buffer += char
        }
        break

      case State.TEXT:
        if (char === ' ') {
          this.tokens.push(this.buffer)
          this.buffer = ''
          this.state = State.START
        }
        else if (char === '{') {
          this.buffer += char
          this.state = State.OPEN_BRACE
        }
        else {
          this.buffer += char
        }
        break

      case State.OPEN_BRACE:
        if (char === '}') {
          this.buffer += char
          this.state = State.POTENTIAL_CLOSE
        }
        else {
          this.buffer += char
        }
        break

      case State.POTENTIAL_CLOSE:
        if (char === '}') {
          this.buffer += char
          this.state = State.BRACES_COMPLETE
        }
        else {
          this.buffer += `}${char}`
          this.state = State.OPEN_BRACE
        }
        break

      case State.BRACES_COMPLETE:
        if (char === ' ') {
          this.tokens.push(this.buffer)
          this.buffer = ''
          this.state = State.START
        }
        else {
          this.buffer += char
          this.state = State.TEXT
        }
        break

      default:
        throw new Error('Unexpected state')
    }
  }

  public run(input: string): string[] {
    for (const char of input) {
      this.processChar(char)
    }
    // Push the last buffer if it's not empty
    if (this.buffer.length > 0) {
      this.tokens.push(this.buffer)
    }
    return this.tokens
  }
}
