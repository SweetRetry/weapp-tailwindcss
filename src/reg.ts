import replace from 'regexp-replace'

export const classRegexp = /(?:class|className)=(?:["']\W+\s*(?:\w+)\()?["']([^"]+)['"]/gim

export const tagRegexp = /<[a-z][-a-z]*[a-z]* *([a-z][-a-z]*[a-z]*(?: *= *"(.*?)")?)* *\/? *>/gim

// /[\r\n\s]*<(?:\/)?([^ =>]+)([^>]*?)(?:\/)?>/gim

// export const noClosedTagRegexp = /[\r\n\s]*<([^ =>]+)([^>]*?)(?:\/)?>/gim

export function classStringReplace (str: string, replacement: (string: string, arr?: RegExpExecArray, index?: number, lastIndex?: number) => string) {
  return replace(str, classRegexp, replacement)
}

export function tagStringRegexp (str: string, replacement: (string: string, arr?: RegExpExecArray, index?: number, lastIndex?: number) => string) {
  return replace(str, tagRegexp, replacement)
}

export const variableRegExp = /{{([^{}]*)}}/g

export function variableMatch (original: string) {
  return variableRegExp.exec(original)
}
