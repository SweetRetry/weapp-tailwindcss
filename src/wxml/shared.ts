import { MappingChars2String as dic } from '@/dic'
import { ICommonReplaceOptions } from '@/types'
import { mangleMark } from '@/mangle/expose'

// export function arbitraryValuesReplacer (str: string) {}

export function replaceWxml (
  original: string,
  options: ICommonReplaceOptions | boolean = {
    keepEOL: false,
    mangle: false
  }
) {
  if (typeof options === 'boolean') {
    options = {
      keepEOL: options,
      mangle: false
    }
  }
  let res = original
    .replace(/\[/g, dic['['])
    .replace(/\]/g, dic[']'])
    .replace(/\(/g, dic['('])
    .replace(/\)/g, dic[')'])
    .replace(/#/g, dic['#']) // hex
    .replace(/!/g, dic['!']) // css !important
    .replace(/\//g, dic['/'])
    .replace(/\./g, dic['.'])
    .replace(/:/g, dic[':'])
    // https://github.com/sonofmagic/weapp-tailwindcss-webpack-plugin/issues/8
    .replace(/%/g, dic['%'])
    .replace(/,/g, dic[','])
    .replace(/\\/g, dic['\\'])
    .replace(/'/g, dic["'"])
    .replace(/"/g, dic['"'])
    .replace(/\*/g, dic['*'])
    .replace(/&/g, dic['&'])
    .replace(/@/g, dic['@'])
    .replace(/{/g, dic['{'])
    .replace(/}/g, dic['}'])
  if (!options.keepEOL) {
    res = res
      // 去除无用换行符和空格
      .replace(/[\r\n]+/g, '')
  }
  if (options.mangle) {
    res = mangleMark(res, options.classGenerator)
  }
  return res
}
