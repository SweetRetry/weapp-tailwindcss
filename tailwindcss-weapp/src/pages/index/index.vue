<script setup lang="ts">
import { devDependencies } from '../../../package.json'
import BaseLayout from '@/components/BaseLayout.vue'
import FloatButton from '@/components/FloatButton.vue'
import { useSystemStore } from '@/stores'

const version = devDependencies.tailwindcss.slice(1)

const store = useSystemStore()

function openSearchBox() {

}

function toggleTheme() {
  if (store.theme === 'dark') {
    store.setTheme('light')
  }
  else {
    store.setTheme('dark')
  }
}

function go2ThemeDemo() {
  uni.navigateTo({
    url: '/pages/theme/index',
  })
}
</script>

<template>
  <BaseLayout>
    <up-navbar fixed bgColor="inherit" safeAreaInsetTop>
      <template #left>
        <view class="relative flex items-center">
          <view class="i-logos-tailwindcss-icon mr-2 h-[20.57px] w-[34px] shrink-0" />
          <view class="text-lg font-semibold">
            Tailwind CSS
          </view>
          <view
            class="relative bottom-0.5 ml-1 self-end rounded-full bg-sky-400/10 px-2 py-0.5 text-xs font-medium text-sky-600"
          >
            {{ version }}
          </view>
        </view>
      </template>
    </up-navbar>
    <FloatButton store-key="index-float-btn" :padding="[256, 32, 256, 32]">
      <view class="float-btn mb-3">
        <view class="i-mdi-flask-round-bottom-empty-outline text-sky-400 dark:text-sky-500" @click="go2ThemeDemo" />
      </view>
      <view class="float-btn mb-3" @click="toggleTheme">
        <view v-if="store.theme === 'dark'" class="i-mdi-white-balance-sunny text-sky-500" />
        <view v-else class="i-mdi-weather-night text-sky-400" />
      </view>
      <view class="float-btn" @click="openSearchBox">
        <view class="i-mdi-magnify text-sky-400 dark:text-sky-500" />
      </view>
    </FloatButton>
  </BaseLayout>
</template>
