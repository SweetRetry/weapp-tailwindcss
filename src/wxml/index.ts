import { parse, render } from 'wxml-ast'
import { parseExpression } from '@babel/parser'
import traverse from '@babel/traverse'
import generate from '@babel/generator'
import { replaceWxml } from './shared'

export { replaceWxml }

// #region internal
const variableRegExp = /{{([^{}]*)}}/g

function variableMatch(original: string) {
  return variableRegExp.exec(original)
}
// #endregion

export function generateCode(match: string) {
  const ast = parseExpression(match)

  traverse(ast, {
    StringLiteral(path) {
      path.node.value = replaceWxml(path.node.value)
    },
    noScope: true
  })

  const { code } = generate(ast, {
    compact: true,
    minified: true
  })
  return code
}

export interface RawSource {
  start: number
  end: number
  raw: string
  // '' 直接 remove {{}}
  source?: string
}

export function templeteReplacer(original: string) {
  let match = variableMatch(original)
  const sources: RawSource[] = []

  while (match !== null) {
    // 过滤空字符串
    // if (match[1].trim().length) {
    sources.push({
      start: match.index,
      end: variableRegExp.lastIndex,
      raw: match[1]
    })

    match = variableMatch(original)
  }

  if (sources.length) {
    const resultArray: string[] = []
    let p = 0
    for (let i = 0; i < sources.length; i++) {
      const m = sources[i]
      // 匹配前值
      resultArray.push(replaceWxml(original.slice(p, m.start), true))
      p = m.start
      // 匹配后值
      if (m.raw.trim().length) {
        const code = generateCode(m.raw)
        m.source = `{{${code}}}`
      } else {
        m.source = ''
      }

      resultArray.push(m.source)
      p = m.end
      // 匹配最终尾部值
      if (i === sources.length - 1) {
        resultArray.push(replaceWxml(original.slice(m.end), true))
      }
    }

    return resultArray.filter((x) => x).join('')
  } else {
    return replaceWxml(original)
  }
}

export async function templeteHandler(rawSource: string) {
  const parsed = await parse(rawSource, {
    elementCB: (node) => {
      if (node.attribs.class) {
        node.attribs.class = templeteReplacer(node.attribs.class)
      }
    }
  })
  return render(parsed)
  // wxml.traverse(parsed, (node, parent) => {
  //   if (node.type === wxml.NODE_TYPES.ELEMENT) {
  //     // @ts-ignore
  //     if (node.attributes.class) {
  //       // @ts-ignore
  //       node.attributes.class = templeteReplacer(node.attributes.class)
  //     }
  //   }
  // })
  // return wxml.serialize(parsed)
}
