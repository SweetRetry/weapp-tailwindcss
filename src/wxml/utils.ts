import * as t from '@babel/types'
import { replaceWxml } from './shared'
import { parseExpression, traverse, generate } from '@/babel'
import { variableMatch, variableRegExp, templateClassExactRegexp, tagWithEitherClassAndHoverClassRegexp, makeCustomAttributes } from '@/reg'
import { defu } from '@/utils'
import type { RawSource, ICommonReplaceOptions, ITempleteHandlerOptions } from '@/types'

export function generateCode(match: string, options: ICommonReplaceOptions = {}) {
  const ast = parseExpression(match)

  traverse(ast, {
    StringLiteral(path) {
      // [g['人生']==='你好啊'?'highlight':'']
      if (t.isMemberExpression(path.parent)) {
        return
      }
      // parentPath maybe null
      // ['td',[(g.type==='你好啊')?'highlight':'']]
      if (t.isBinaryExpression(path.parent) && t.isConditionalExpression(path.parentPath?.parent)) {
        return
      }

      path.node.value = replaceWxml(path.node.value, options)
    },
    noScope: true
  })
  // @ts-ignore
  const { code } = generate(ast, {
    compact: true,
    minified: true,
    jsescOption: {
      quotes: 'single',
      minimal: true
    }
  })

  return code
}

export function extractSource(original: string) {
  let match = variableMatch(original)
  const sources: RawSource[] = []

  while (match !== null) {
    // 过滤空字符串
    // if (match[1].trim().length) {
    const start = match.index
    const end = variableRegExp.lastIndex
    sources.push({
      start,
      end,
      raw: match[1],
      prevConcatenated: !/\s/.test(original[start - 1]),
      nextConcatenated: !/\s/.test(original[end])
    })

    match = variableMatch(original)
  }
  return sources
}

export function templeteReplacer(original: string, options: ICommonReplaceOptions = {}) {
  const sources = extractSource(original)

  if (sources.length > 0) {
    const resultArray: string[] = []
    let p = 0
    for (let i = 0; i < sources.length; i++) {
      const m = sources[i]
      const before = original.slice(p, m.start)

      // 匹配前值
      resultArray.push(
        replaceWxml(before, {
          keepEOL: true,
          escapeMap: options.escapeMap
        })
      )
      p = m.start
      // 匹配后值
      if (m.raw.trim().length > 0) {
        const code = generateCode(m.raw, options)
        const source = `{{${code}}}`
        m.source = source
      } else {
        m.source = ''
      }

      resultArray.push(m.source)
      p = m.end
      // 匹配最终尾部值
      if (i === sources.length - 1) {
        const after = original.slice(m.end)
        resultArray.push(
          replaceWxml(after, {
            keepEOL: true,
            escapeMap: options.escapeMap
          })
        )
      }
    }

    return resultArray
      .filter(Boolean)
      .join('')
      .trim()
  } else {
    return replaceWxml(original, {
      keepEOL: false,
      escapeMap: options.escapeMap
    })
  }
}

export function templeteHandler(rawSource: string, options: ICommonReplaceOptions = {}) {
  return rawSource.replace(tagWithEitherClassAndHoverClassRegexp, (m0) => {
    return m0.replace(templateClassExactRegexp, (m1, className) => {
      return m1.replace(className, templeteReplacer(className, options))
    })
  })
}

export function customTempleteHandler(rawSource: string, options: ITempleteHandlerOptions = {}) {
  let source = templeteHandler(rawSource, options)
  const regexps = makeCustomAttributes(options.customAttributesEntities)
  if (regexps) {
    if (Array.isArray(regexps)) {
      for (const regexp of regexps) {
        source = source.replace(regexp.tagRegexp, (m0) => {
          return m0.replace(regexp.attrRegexp, (m1, className) => {
            return m1.replace(className, templeteReplacer(className, options))
          })
        })
      }
    }

    return source
  } else {
    return source
  }
}

export function createTempleteHandler(options: ITempleteHandlerOptions = {}) {
  return (rawSource: string, opt: ITempleteHandlerOptions = {}) => {
    return customTempleteHandler(rawSource, defu(opt, options))
  }
}
