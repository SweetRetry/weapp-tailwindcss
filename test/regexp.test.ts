// import replace from 'regexp-replace'
import { classStringReplace, tagStringRegexp } from '@/reg'
import { replaceWxml } from '@/wxml/index'
describe('regexp', () => {
  test('tagStringRegexp', async () => {
    const wxmlCase =
      '<view class="p-[20px] -mt-2 mb-[-20px]">p-[20px] -mt-2 mb-[-20px] margin的jit 不能这么写 -m-[20px]</view><view class="space-y-[1.6rem]"><view class="w-[300rpx] text-black text-opacity-[0.19]">w-[300rpx] text-black text-opacity-[0.19]</view><view class="min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]">min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]</view><view class="max-w-[300rpx] min-h-[100px] text-[#dddddd]">max-w-[300rpx] min-h-[100px] text-[#dddddd]</view><view class="flex items-center justify-center h-[100px] w-[100px] rounded-[40px] bg-[#123456] bg-opacity-[0.54] text-[#ffffff]">Hello</view><view class="border-[10px] border-[#098765] border-solid border-opacity-[0.44]">border-[10px] border-[#098765] border-solid border-opacity-[0.44]</view><view class="grid grid-cols-3 divide-x-[10px] divide-[#010101] divide-solid"><view>1</view><view>2</view><view>3</view></view></view><view class="test">test</view>'

    const str = tagStringRegexp(wxmlCase, (x) => {
      const res = classStringReplace(x, (y) => {
        return replaceWxml(y)
      })
      return res
    })

    expect(str).toBe(
      '<view class="p-_l_20px_r_ -mt-2 mb-_l_-20px_r_">p-[20px] -mt-2 mb-[-20px] margin的jit 不能这么写 -m-[20px]</view><view class="space-y-_l_1-dot-6rem_r_"><view class="w-_l_300rpx_r_ text-black text-opacity-_l_0-dot-19_r_">w-[300rpx] text-black text-opacity-[0.19]</view><view class="min-w-_l_300rpx_r_ max-h-_l_100px_r_ text-_l_20px_r_ leading-_l_0-dot-9_r_">min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]</view><view class="max-w-_l_300rpx_r_ min-h-_l_100px_r_ text-_l__h_dddddd_r_">max-w-[300rpx] min-h-[100px] text-[#dddddd]</view><view class="flex items-center justify-center h-_l_100px_r_ w-_l_100px_r_ rounded-_l_40px_r_ bg-_l__h_123456_r_ bg-opacity-_l_0-dot-54_r_ text-_l__h_ffffff_r_">Hello</view><view class="border-_l_10px_r_ border-_l__h_098765_r_ border-solid border-opacity-_l_0-dot-44_r_">border-[10px] border-[#098765] border-solid border-opacity-[0.44]</view><view class="grid grid-cols-3 divide-x-_l_10px_r_ divide-_l__h_010101_r_ divide-solid"><view>1</view><view>2</view><view>3</view></view></view><view class="test">test</view>'
    )
  })
})
