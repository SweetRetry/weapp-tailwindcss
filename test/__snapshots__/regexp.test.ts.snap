// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regexp static node  1`] = `"<view class=\\"p-_bl_20px_br_ -mt-2 mb-_bl_-20px_br_\\">p-[20px] -mt-2 mb-[-20px] margin的jit 不能这么写 -m-[20px]</view><view class=\\"space-y-_bl_1_d_6rem_br_\\"><view class=\\"w-_bl_300rpx_br_ text-black text-opacity-_bl_0_d_19_br_\\">w-[300rpx] text-black text-opacity-[0.19]</view><view class=\\"min-w-_bl_300rpx_br_ max-h-_bl_100px_br_ text-_bl_20px_br_ leading-_bl_0_d_9_br_\\">min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]</view><view class=\\"max-w-_bl_300rpx_br_ min-h-_bl_100px_br_ text-_bl__h_dddddd_br_\\">max-w-[300rpx] min-h-[100px] text-[#dddddd]</view><view class=\\"flex items-center justify-center h-_bl_100px_br_ w-_bl_100px_br_ rounded-_bl_40px_br_ bg-_bl__h_123456_br_ bg-opacity-_bl_0_d_54_br_ text-_bl__h_ffffff_br_\\">Hello</view><view class=\\"border-_bl_10px_br_ border-_bl__h_098765_br_ border-solid border-opacity-_bl_0_d_44_br_\\">border-[10px] border-[#098765] border-solid border-opacity-[0.44]</view><view class=\\"grid grid-cols-3 divide-x-_bl_10px_br_ divide-_bl__h_010101_br_ divide-solid\\"><view>1</view><view>2</view><view>3</view></view></view><view class=\\"test\\">test</view>"`;

exports[`regexp static node = self 1`] = `"<view class=\\"p-[20px] -mt-2 mb-[-20px]\\">p-[20px] -mt-2 mb-[-20px] margin的jit 不能这么写 -m-[20px]</view><view class=\\"space-y-[1.6rem]\\"><view class=\\"w-[300rpx] text-black text-opacity-[0.19]\\">w-[300rpx] text-black text-opacity-[0.19]</view><view class=\\"min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]\\">min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]</view><view class=\\"max-w-[300rpx] min-h-[100px] text-[#dddddd]\\">max-w-[300rpx] min-h-[100px] text-[#dddddd]</view><view class=\\"flex items-center justify-center h-[100px] w-[100px] rounded-[40px] bg-[#123456] bg-opacity-[0.54] text-[#ffffff]\\">Hello</view><view class=\\"border-[10px] border-[#098765] border-solid border-opacity-[0.44]\\">border-[10px] border-[#098765] border-solid border-opacity-[0.44]</view><view class=\\"grid grid-cols-3 divide-x-[10px] divide-[#010101] divide-solid\\"><view>1</view><view>2</view><view>3</view></view></view><view class=\\"test\\">test</view>"`;

exports[`regexp tagStringReplace 1`] = `"<view class=\\"p-_bl_20px_br_ -mt-2 mb-_bl_-20px_br_\\">p-[20px] -mt-2 mb-[-20px] margin的jit 不能这么写 -m-[20px]</view><view class=\\"space-y-_bl_1_d_6rem_br_\\"><view class=\\"w-_bl_300rpx_br_ text-black text-opacity-_bl_0_d_19_br_\\">w-[300rpx] text-black text-opacity-[0.19]</view><view class=\\"min-w-_bl_300rpx_br_ max-h-_bl_100px_br_ text-_bl_20px_br_ leading-_bl_0_d_9_br_\\">min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]</view><view class=\\"max-w-_bl_300rpx_br_ min-h-_bl_100px_br_ text-_bl__h_dddddd_br_\\">max-w-[300rpx] min-h-[100px] text-[#dddddd]</view><view class=\\"flex items-center justify-center h-_bl_100px_br_ w-_bl_100px_br_ rounded-_bl_40px_br_ bg-_bl__h_123456_br_ bg-opacity-_bl_0_d_54_br_ text-_bl__h_ffffff_br_\\">Hello</view><view class=\\"border-_bl_10px_br_ border-_bl__h_098765_br_ border-solid border-opacity-_bl_0_d_44_br_\\">border-[10px] border-[#098765] border-solid border-opacity-[0.44]</view><view class=\\"grid grid-cols-3 divide-x-_bl_10px_br_ divide-_bl__h_010101_br_ divide-solid\\"><view>1</view><view>2</view><view>3</view></view></view><view class=\\"test\\">test</view>"`;

exports[`regexp tagStringReplace2 1`] = `"<view class=\\"p-_bl_20px_br_ -mt-2 mb-_bl_-20px_br_\\">p-[20px] -mt-2 mb-[-20px] margin的jit 不能这么写 -m-[20px]</view><view class=\\"space-y-_bl_1_d_6rem_br_\\"><view class=\\"w-_bl_300rpx_br_ text-black text-opacity-_bl_0_d_19_br_\\">w-[300rpx] text-black text-opacity-[0.19]</view><view class=\\"min-w-_bl_300rpx_br_ max-h-_bl_100px_br_ text-_bl_20px_br_ leading-_bl_0_d_9_br_\\">min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]</view><view class=\\"max-w-_bl_300rpx_br_ min-h-_bl_100px_br_ text-_bl__h_dddddd_br_\\">max-w-[300rpx] min-h-[100px] text-[#dddddd]</view><view class=\\"flex items-center justify-center h-_bl_100px_br_ w-_bl_100px_br_ rounded-_bl_40px_br_ bg-_bl__h_123456_br_ bg-opacity-_bl_0_d_54_br_ text-_bl__h_ffffff_br_\\">Hello</view><view class=\\"border-_bl_10px_br_ border-_bl__h_098765_br_ border-solid border-opacity-_bl_0_d_44_br_\\">border-[10px] border-[#098765] border-solid border-opacity-[0.44]</view><view class=\\"grid grid-cols-3 divide-x-_bl_10px_br_ divide-_bl__h_010101_br_ divide-solid\\"><view>1</view><view>2</view><view>3</view></view></view><view class=\\"test\\">test</view>"`;
